### DISCUSSION TITLE
Grouping duplicate keys in configuration

### DISCUSSION THREAD

**Discussion by fmunteanu** [renovatebot/renovate#28992] [Request Help] [answered]
### What would you like help with?

I would like help with my configuration

### How are you running Renovate?

Mend Renovate hosted app on github.com

### Please tell us more about your question or problem

This is related to #28942, where @rarkins provided useful insight.

The [renovate.json](https://github.com/axivo/renovate-test/blob/main/.github/renovate.json) file will become quite large and I was wondering if I could group the duplicate keys, with a similar approach to `matchStringsStrategy`. Technically, I would like to avoid repeating keys like `customType`, `datasourceTemplate` and `versioningTemplate`.

Thank you for your help.

### Logs (if relevant)

<details><summary>Logs</summary>

```

Replace this text with your logs, between the starting and ending triple backticks

```

</details>

**Comment by rarkins** [comment, marked as answer]
You can probably drop versioningTemplate because Renovate will use the default versioning for the datasource. customType defaults to regex but I think we made the choice that we want it configured explicitly. You can't avoid configuring datasourceTemplate, especially as you have a mix of those

**Reply by fmunteanu** [reply]
I removed all versioning templates, except the K3s `"versioningTemplate": "semver-coerced"` one.
